src/Handler/TagHandler.php

```diff
@@ -40,18 +45,20 @@ class TagHandler
     /**
      * Constructor
      *
-     * @param CacheInvalidator $invalidator The invalidator instance.
-     * @param string           $tagsHeader  Header to use for tags, defaults to X-Cache-Tags.
+     * @param CacheInvalidator $invalidator  The invalidator instance.
+     * @param string           $tagsHeader   Header to use for tags, defaults to X-Cache-Tags.
+     * @param int              $headerLength Maximum header size in bytes, defaults to 7500.
      *
      * @throws UnsupportedProxyOperationException If CacheInvalidator does not support invalidate requests
      */
-    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags')
+    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags', $headerLength = 7500)
```

Comment:
my plan is to create a branch update-master from 1.4, merge master into it and solve conflicts. as part of that we would shuffle things around as needed. this amounts to about the same work as doing a separate PR, but has the benefit that git understands in the future how to merge 1.4 into master again.